### PR TITLE
platform: add Platform method

### PR DESCRIPTION
**Pull request by mischief**

### CODE REVIEW

File: platform/aws.go [hunk 1]
```diff
@@ -32,6 +32,12 @@ type awsMachine struct {
 	mach    *ec2.Instance
 }
 
+var _ Machine = (*awsMachine)(nil)
```
Comment (marineam) [inline platform/aws.go]: What is the point of this?

Comment (mischief) [inline platform/aws.go]: it asserts awsMachine implements Machine at build time.

Comment (marineam) [inline platform/aws.go]: Ohh, is this an assertion that it provides the interface? But doesn't NewMachine provide that assertion?

Comment (mischief) [inline platform/aws.go]: i suppose it does, this is simply more explicit.

Comment (marineam) [inline platform/aws.go]: Considering Go tends to not use explicit assertions I think the implicit one in NewMachine is perfectly sufficient.

**Comment by marineam** [comment]
Update the commit message since the assertion is gone, but LGTM

**Pull request closed without merging**